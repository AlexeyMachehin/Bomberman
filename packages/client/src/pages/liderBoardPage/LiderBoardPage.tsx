import styles from './LiderBoardPage.module.css';

import Container from '@mui/material/Container';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';
import ListItemAvatar from '@mui/material/ListItemAvatar';
import ListItemText from '@mui/material/ListItemText';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import AccountCircleIcon from '@mui/icons-material/AccountCircle';
import StarBorderIcon from '@mui/icons-material/StarBorder';
import EmojiEventsIcon from '@mui/icons-material/EmojiEvents';

const liders = [
  {
    id: 1,
    name: 'Иван Петров',
    rating: 1235,
    avatar: 'https://placekitten.com/100',
  },
  {
    id: 2,
    name: 'Петр Иванов',
    rating: 1135,
    avatar: 'https://placekitten.com/100',
  },
  {
    id: 3,
    name: 'Арсений Сидоров',
    rating: 235,
  },
  {
    id: 4,
    name: 'Вениамин Скачков',
    rating: 135,
    avatar: 'https://placekitten.com/100',
  },
];

const Colors = ['gold', 'silver', 'goldenrod', 'tan'];

const getIcon = (index: number) => {
  const bestLidersCount = 3;
  return index < bestLidersCount ? (
    <EmojiEventsIcon fontSize="large" />
  ) : (
    <StarBorderIcon fontSize="medium" />
  );
};

const getColor = (index: number) => {
  return Colors[index] ?? Colors[-1];
};

const LiderBoardPage = () => {
  return (
    <div className={styles.liderBoard}>
      <Container maxWidth="lg">
        <h1 className={styles.title}>LiderboardPage</h1>
        <List>
          {liders.map((lider, index) => (
            <ListItem key={lider.id} className={styles.listItem}>
              <ListItemAvatar>
                <Avatar src={lider.avatar} alt={lider.name}>
                  <AccountCircleIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText>{lider.name}</ListItemText>
              <ListItemText sx={{marginLeft: 'auto', flexGrow: '0'}}>
                <Typography
                  display="flex"
                  alignItems="center"
                  color={getColor(index)}>
                  {lider.rating}
                  {getIcon(index)}
                </Typography>
              </ListItemText>
            </ListItem>
          ))}
        </List>
      </Container>
    </div>
  );
};

export default LiderBoardPage;
