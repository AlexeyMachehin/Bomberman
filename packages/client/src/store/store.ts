import { configureStore } from '@reduxjs/toolkit';
import { userSlice } from './user/userSlice';
import { forumSlice } from './forum/forumSlice';

export const createStore = () => {
  return configureStore({
    reducer: {
      userReducer: userSlice.reducer,
      forumReducer: forumSlice.reducer,
    },
  });
};

export const store = createStore();

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
