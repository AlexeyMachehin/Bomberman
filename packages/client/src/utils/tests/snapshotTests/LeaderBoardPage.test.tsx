import LiderBoardPage from '@/pages/leaderBoardPage/LeaderBoardPage';
import { render } from '@testing-library/react';
import { store } from '@/store/store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

test('LeaderBoardPage renders correctly', () => {
  const component = render(
    <MemoryRouter>
      <Provider store={store}>
        <LiderBoardPage />
      </Provider>
    </MemoryRouter>
  );
  expect(component).toMatchSnapshot();
});
