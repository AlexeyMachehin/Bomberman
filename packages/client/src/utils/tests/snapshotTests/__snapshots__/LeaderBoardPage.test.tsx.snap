// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoardPage renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="leaderBoardPage-component"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
        >
          <div>
            <h1>
              LeaderboardPage
            </h1>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FirstPageIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                  />
                </svg>
              </span>
              start page
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="leaderBoardPage-component"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div>
          <h1>
            LeaderboardPage
          </h1>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FirstPageIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18.41 16.59 13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                />
              </svg>
            </span>
            start page
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
